tests/migration/online_migration/yast2_migration.pm

```diff
@@ -209,18 +209,25 @@ sub run {
     send_key "alt-p";    # focus on the item of possible migration targets
     assert_screen 'yast2-migration-target-list-selected', 60;
     send_key_until_needlematch 'migration-target-' . get_var("VERSION"), 'down', 20, 3;
-    send_key "alt-n";
+
+    # On some platforms such as Aarch64 and x86_64, after sending alt-n keys,
+    # we need to send_key alt-n more than once for the next screen to show up.
+    my $wait_time = 720;
+    my @next_tags = qw(import-untrusted-gpg-key yast2-migration-installupdate yast2-migration-proposal);
+    if (get_var('SMT_URL') =~ /smt/) {
+        $wait_time = 60;
+    }
+    send_key_until_needlematch \@next_tags, 'alt-n', 5, $wait_time;
+
     # migration via smt will install packagehub and NVIDIA compute, we need click trust
     # gpg keys; Same with leap to sle migration, need to trust packagehub gpg key.
-    if (get_var('SMT_URL') =~ /smt/) {
-        assert_screen 'import-untrusted-gpg-key', 60;
+    if (get_var('SMT_URL') =~ /smt/ && (match_has_tag 'import-untrusted-gpg-key')) {
```

Comment:
Do you need extend 'match_has_tag' timeout time?

Reply:
added. thanks a lot

Reply:
> Do you need extend 'match_has_tag' timeout time?

There's no timeout parameter for this funciton.

Reply:
`match_has_tag` checks if the last match has a certain time. For example in the code

```perl
assert_screen([qw(generic-desktop generic-desktop-with-bug)]);
die "Found bug" if match_has_tag 'generic-desktop-with-bug';
```

here `assert_screen` has a default timeout of up to 30s. After that line, if `assert_screen` has found a needle with at least one of both tags then the next line with `match_has_tag` is called which returns immediately with a boolean result. As a needle can have multiple tags it can also be used like this

```perl
assert_screen 'generic-desktop';
die "Found bug" if match_has_tag 'generic-desktop-with-bug';
```

The die-call in the second line will be executed if a needle is found that has both the tag "generic-desktop", so being evaluated in the `assert_screen` call as well as having the tag "generic-desktop-with-bug" so that `match_has_tag` returns true in this case